twitter-clone/src/pages/Dashboard/components/NewTweetForm.tsx

```diff
@@ -0,0 +1,72 @@
+import { useContext, useState } from 'react';
+import sanitizeHtml from 'sanitize-html';
+
+import { Button, Container } from '@/components';
+import { useUser } from '@/hooks';
+import { tryCatch } from '@/utils';
+import { Tweet } from '@/types';
+import { TweetContext } from '@/contexts';
+import { submitTweet } from '@/api';
+
+export default function NewTweetForm() {
+  const [tweetContent, setTweetContent] = useState<string>('');
+  const [submitStatus, setSubmitStatus] = useState<string>('');
+
+  const { tweets: currentTweets, setTweets } = useContext(TweetContext);
+
+  const { user } = useUser();
+
+  function isFieldValid(): boolean {
+    return tweetContent.length >= 1 && tweetContent.length <= 140;
+  }
+
+  async function handleSubmitTweet() {
```

Comment:
It could be a good idea to move it to separated hook and add some loading state.